tests/testthat/test-prepare_pctr_product.R

```diff
@@ -98,3 +98,36 @@ test_that("handles tiltIndicator output", {
     )
   )
 })
+
+test_that("yields a single distinct value of `*matching_certainty_company_average` per company", {
+  id <- "id3"
+  clustered_one <- "alarm system"
+  clustered_two <- "aluminium"
+  uuid_one <- "3d731062-1960-5a36-bd19-6ab2b0bf67c2_245732f4-a5ce-4881-816b-a207ba8df4c8; 3b11f419-2204-5b7c-8cf5-70650f0555ca_245732f4-a5ce-4881-816b-a207ba8df4c8"
+  uuid_two <- "ff4fd9d9-7dcb-5a50-926f-76ae31fa454d_618bf4eb-bee5-4d38-8e4f-78cfebf779be; 73f37024-063b-507c-a976-5bc8e1ee469f_618bf4eb-bee5-4d38-8e4f-78cfebf779be"
+
+  product <- tibble(
+    companies_id = id,
+    grouped_by = "a",
+    risk_category = "a",
+    clustered = c(clustered_one, clustered_one, clustered_two),
+    activity_uuid_product_uuid = c(uuid_one, uuid_one, uuid_two),
+    co2_footprint = "a",
+    tilt_sector = "a",
+    tilt_subsector = "a",
+    isic_4digit = "a"
+  )
+
+  result <- prepare_pctr_product(
+    product,
+    ep_companies,
+    ecoinvent_activities,
+    small_matches_mapper,
+    isic_tilt_mapper
+  )
+  out <- result |>
+    group_by(companies_id) |>
+    summarise(count = n_distinct(matching_certainty_company_average))
```

Comment:
Same as above, if we're using a single company then `group_by(companies_id)` is unnecessary. Instead we should use a simpler expectation.

``` r
devtools::load_all()
#> ℹ Loading tiltIndicatorAfter

id <- "id3"
clustered_one <- "alarm system"
clustered_two <- "aluminium"
uuid_one <- "3d731062-1960-5a36-bd19-6ab2b0bf67c2_245732f4-a5ce-4881-816b-a207ba8df4c8; 3b11f419-2204-5b7c-8cf5-70650f0555ca_245732f4-a5ce-4881-816b-a207ba8df4c8"
uuid_two <- "ff4fd9d9-7dcb-5a50-926f-76ae31fa454d_618bf4eb-bee5-4d38-8e4f-78cfebf779be; 73f37024-063b-507c-a976-5bc8e1ee469f_618bf4eb-bee5-4d38-8e4f-78cfebf779be"

product <- tibble(
  companies_id = id,
  grouped_by = "a",
  risk_category = "a",
  clustered = c(clustered_one, clustered_one, clustered_two),
  activity_uuid_product_uuid = c(uuid_one, uuid_one, uuid_two),
  co2_footprint = "a",
  tilt_sector = "a",
  tilt_subsector = "a",
  isic_4digit = "a"
)

result <- prepare_pctr_product(
  product,
  ep_companies,
  ecoinvent_activities,
  small_matches_mapper,
  isic_tilt_mapper
)

expect_equal(unique(result$matching_certainty_company_average), "low")
```

<sup>Created on 2023-10-17 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
Resolved :) Thanks